qiskit/transpiler/passes/scheduling/asap.py

```diff
@@ -128,27 +120,20 @@ def run(self, dag):
                     t0 = max(idle_after[bit] for bit in node.qargs + node.cargs)
                     t1 = t0 + op_duration
 
-            # Add delay to qubit wire
             for bit in node.qargs:
-                delta = t0 - idle_after[bit]
-                if delta > 0 and isinstance(bit, Qubit):
-                    new_dag.apply_operation_back(Delay(delta, time_unit), [bit], [])
                 idle_after[bit] = t1
 
-            new_dag.apply_operation_back(node.op, node.qargs, node.cargs)
+            # Place the node in time slot. Ignore delay since this is identical to empty.
+            if not isinstance(node.op, Delay):
+                # Note that some transformation pass can override .op attribute of DAGOpNode
+                # without breaking the reference to object. However, once operation is updated,
+                # there is no guarantee that the duration of node is unchanged.
+                key = node.name, node.sort_key, getattr(node, "_node_id")
+                node_start_time[key] = t0
```

Comment:
I understand your point. This code is indeed not robust to the future change of node data structure. I guess the mutability of node.op is due to performance of the `Unroller` node, which is likely intentional and a pass manager designer must take care for the arrangement of passes. In that sense, the padding pass here can also take node as-is rather than decomposing it for the key, and let the designer carefully choose the order of passes.